src/lib/signature_lib/schnorr.ml

```diff
@@ -444,14 +498,18 @@ module Message = struct
 
   type t = (Field.t, bool) Random_oracle.Input.t [@@deriving sexp]
 
+  let network_id_mainnet = Char.of_int_exn 1
+
+  let network_id_testnet = Char.of_int_exn 0
+
   let network_id =
     match Mina_signature_kind.t with
     | Mainnet ->
-        Char.of_int_exn 1
+        network_id_mainnet
     | Testnet ->
-        Char.of_int_exn 0
+        network_id_testnet
 
-  let derive t ~private_key ~public_key =
```

Comment:
I don't think this is a good approach, if you sign the same transaction on both networks then you will derive different nonces here (for the same message and private key), which is not what this algorithm has in mind. I think a better approach would be to include the chain id, or network id, in the data being signed (e.g. a transaction).